internal/agent/install.go

```diff
@@ -113,6 +113,10 @@ func Install(dir ...string) error {
 		if exists {
 			tmpdir, err := os.MkdirTemp("", "kairos-install-")
 			if err == nil {
+				// Add cloud config header if necessary
+				if !strings.Contains(cloudConfig, "#cloud-config") {
```

Comment:
Maybe we should adapt and use this: https://github.com/kairos-io/kairos-agent/blob/9f46af255f056c974d335dd305eac837fb012efb/pkg/config/config.go#L273

Reply:
This is not a Config object so we cannot access that. This is just a plain string.